integration_tests/test_solo_machine.py

```diff
@@ -0,0 +1,179 @@
+        self.trusted_hash = result["result"]["block_id"]["hash"]
+        self.trusted_height = result["result"]["block"]["header"]["height"]
+
+    def set_env(self):
+        os.environ["SOLO_DB_URI"] = self.db_path
+        os.environ["SOLO_SIGNER"] = self.sign_file
+        os.environ[
+            "SOLO_MNEMONIC"
+        ] = "awesome there minute cash write immune tag \
+reopen price congress trouble reunion south wisdom donate credit below leave \
+wisdom eagle sail siege rice train"
+        os.environ["SOLO_HD_PATH"] = "m/44'/394'/0'/0/0"
+        os.environ["SOLO_ACCOUNT_PREFIX"] = "cro"
+        os.environ["SOLO_ADDRESS_ALGO"] = "secp256k1"
+        os.environ["SOLO_FEE_AMOUNT"] = "1000"
+        os.environ["SOLO_FEE_DENOM"] = "basecro"
+        os.environ["SOLO_GAS_LIMIT"] = "300000"
+        os.environ["SOLO_GRPC_ADDRESS"] = f"http://0.0.0.0:{self.grpc_port}"
+        os.environ["SOLO_RPC_ADDRESS"] = f"http://0.0.0.0:{self.rpc_port}"
+        os.environ["SOLO_TRUSTED_HASH"] = self.trusted_hash
+        os.environ["SOLO_TRUSTED_HEIGHT"] = self.trusted_height
+
+    def prepare(self):
+        self.get_chain_info()
+        self.set_env()
+
+    def run_sub_cmd(self, sub_cmd, json_output=True):
+        if json_output:
+            cmd = f"{self.bin_file} --output json {sub_cmd}"
+        else:
+            cmd = f"{self.bin_file} {sub_cmd}"
+        output = interact(cmd)
+        data = parse_output(output, json_output)
+        return data
+
+    def init(self):
+        sub_cmd = "init"
+        return self.run_sub_cmd(sub_cmd)
+
+    def add_chain(self):
+        sub_cmd = "chain add"
+        return self.run_sub_cmd(sub_cmd)
+
+    def connect_chain(self):
+        sub_cmd = f"ibc connect {self.chain_id}"
+        result = self.run_sub_cmd(sub_cmd, False)
+        if "error" in result:
+            raise Exception(result)
+
+    def get_balance(self):
+        sub_cmd = f"chain balance {self.chain_id} solotoken"
+        return self.run_sub_cmd(sub_cmd)["data"]["balance"]
+
+    # fixme after this PR merged, currently only support hex string amount
+    # https://github.com/crypto-com/ibc-solo-machine/pull/82
+    def mint(self, amount="0x20"):
+        sub_cmd = f"ibc mint {self.chain_id} {amount} solotoken"
+        return self.run_sub_cmd(sub_cmd)
+
+    # fixme fix after this PR merged, currently only support hex string amount
+    # https://github.com/crypto-com/ibc-solo-machine/pull/82
+    def burn(self, amount="0x10"):
+        sub_cmd = f"ibc burn {self.chain_id} {amount} solotoken"
+        return self.run_sub_cmd(sub_cmd)
+
+
+def test_solo_machine(cluster, tmpdir_factory):
+    """
+    check solo machine
+    """
+    cli = cluster
+    wait_for_block(cli, 1)
+
+    # get the chain balance
+    chain_solo_addr = cluster.address("solo-signer")
+    chain_balance_1 = cluster.balance(chain_solo_addr)
+    assert chain_balance_1 == 1500 * 10 ** 8
```

Comment:
the magic number can be extracted into a constant?  e.g. `CRO_DECIMALS = 10 ** 8` unless it's already defined somewhere else?